### DISCUSSION TITLE
replace nodata with 0s with elasticsearch when there is actually nothing in database in select time range

### DISCUSSION THREAD

**Discussion by QTSH02** [grafana/grafana#68436] [Alerting]
I use es to spot some error and recover messages, and send them out with labels to represent which IDs had entered false states, but if I use the group by function, and there is no data in the recovery query within the time range, the alert won't be triggered

![screenshot-20230515-144054](https://github.com/grafana/grafana/assets/93812067/e1b80796-e3aa-49cf-ba00-4e0d5e71937f)
like in this screenshot, the top query is used to monitor the error state, and the bottom is used to monitor recover message, but with no data at bottom query, the alert won't be triggered.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82699. Feel free to continue the discussion around this there. Thank you!
